controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -107,237 +107,70 @@ func ReadEnv() string {
-			}
-		}
-
-		if statErr := r.Client.Status().Update(ctx, &env); statErr != nil {
-			log.Info("Namespace create error", "err", statErr)
-			if setClowdStatusErr := SetClowdEnvConditions(ctx, r.Client, &env, crd.ReconciliationFailed, statErr); setClowdStatusErr != nil {
-				log.Info("Set status error", "err", setClowdStatusErr)
-				return ctrl.Result{Requeue: true}, setClowdStatusErr
-			}
-			return ctrl.Result{Requeue: true}, statErr
-		}
-	}
-
-	ens := &core.Namespace{}
-	if getNSErr := r.Client.Get(ctx, types.NamespacedName{Name: env.Status.TargetNamespace}, ens); getNSErr != nil {
-		return ctrl.Result{Requeue: true}, getNSErr
-	}
-
-	if ens.ObjectMeta.DeletionTimestamp != nil {
-		log.Info("Env target namespace is to be deleted - skipping reconcile")
-		return ctrl.Result{}, nil
-	}
-
-	SetEnv(env.Name)
-	defer ReleaseEnv()
-	provErr := runProvidersForEnv(log, provider)
-
-	if provErr != nil {
-		log.Info("Prov err", "err", provErr)
-		if setClowdStatusErr := SetClowdEnvConditions(ctx, r.Client, &env, crd.ReconciliationFailed, provErr); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
-		}
-		return ctrl.Result{Requeue: true}, provErr
-	}
-
-	cacheErr := cache.ApplyAll()
-
-	if cacheErr != nil {
-		log.Info("Cache error", "err", cacheErr)
-		if setClowdStatusErr := SetClowdEnvConditions(ctx, r.Client, &env, crd.ReconciliationFailed, cacheErr); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
-		}
-		return ctrl.Result{Requeue: true}, cacheErr
-	}
-
-	if setAppErr := r.setAppInfo(provider); setAppErr != nil {
-		log.Info("setAppInfo error", "err", setAppErr)
-		if setClowdStatusErr := SetClowdEnvConditions(ctx, r.Client, &env, crd.ReconciliationFailed, setAppErr); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
-		}
-		return ctrl.Result{Requeue: true}, setAppErr
-	}
-
-	if statusErr := SetEnvResourceStatus(ctx, r.Client, &env); statusErr != nil {
-		log.Info("SetEnvResourceStatus error", "err", statusErr)
-		if setClowdStatusErr := SetClowdEnvConditions(ctx, r.Client, &env, crd.ReconciliationFailed, statusErr); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
-		}
-		return ctrl.Result{Requeue: true}, statusErr
-	}
-
-	setPrometheusStatus(&env)
-
-	envReady, _, getEnvResErr := GetEnvResourceStatus(ctx, r.Client, &env)
-	if getEnvResErr != nil {
-		log.Info("GetEnvResourceStatus error", "err", getEnvResErr)
-		return ctrl.Result{Requeue: true}, getEnvResErr
-	}
-
-	envStatus := core.ConditionFalse
-	successCond := cond.Get(&env, crd.ReconciliationSuccessful)
-	if successCond != nil {
-		envStatus = successCond.Status
-	}
+	r.initMetrics(env)
```

Comment:
Is there a reason that metrics is handled differently here?

Reply:
To try and match the original implementation as closely as possible. Metrics are initialized in the clowd environment far earlier in the Reconcile method than clowd app, before the point where the step system takes over. In clowd app the metrics are initialized in the middle of the process. I'm happy to put it wherever you suggest but this was me trying to preserve as much of the original logic flow as I could.

Reply:
Discussed. Leaving it as is.